extensions/ql-vscode/src/cli.ts

```diff
@@ -210,15 +216,23 @@ export class CodeQLCliServer implements Disposable {
    * Launch the cli server
    */
   private async launchProcess(): Promise<child_process.ChildProcessWithoutNullStreams> {
-    const config = await this.getCodeQlPath();
-    return spawnServer(
-      config,
+    const codeQlPath = await this.getCodeQlPath();
+    const server = await spawnServer(
+      codeQlPath,
       'CodeQL CLI Server',
       ['execute', 'cli-server'],
       [],
       this.logger,
       _data => { /**/ }
     );
+
+    if (codeQlPath !== this.codeQlPath) {
+      // path has changed. Update the version info
+      this.codeQlPath = codeQlPath;
+      this.version = await getCodeQlCliVersion(codeQlPath, this.logger);
```

Comment:
I see why this works, but my instinct is that we already have this information, and we don't need to recompute it:

Specifically, `this.getCodeQlPath()` could be made more general, so that it returns the entire `Distribution` object instead of just the path string. The `DistributionProvider` here already knows the CLI version, and whether it is compatible with the extension, it just needs to expose that information as part of the interface instead of being hidden in `DistributionManager`.
Then, instead of making another CLI call to get the version here, we just compare the property.

Reply:
That's not always true. In `ExtensionSpecificDistributionManager.getCodeQlPathWithoutVersionCheck`, the path is grabbed by joining a few strings and directory names. There's no version info there that I can find. Unless the version is hidden in the directory somewhere, I don't know how we can do it.

Reply:
I see, since we've deliberately omitted the version check in that case. Any chance we can call `DistributionManager.getDistribution() instead`? That already knows how to look up the version.

Reply:
We can do that. Though, internally, `DistributionManager.getDistribution()` uses `getCodeQlCliVersion`. It's maybe a bit cleaner this way. Implemented, tested locally and pushed.